File: Julia/problem004.jl
#=
Problem 4 - Project Euler
http://projecteuler.net/index.php?section=problems&id=4
=#

is_palindromic(x) = string(x) == reverse(string(x))

function search()
    pmax = 0
    for i in 100:999
        for j in i:999
            p = i * j
            if is_palindromic(p) && p > pmax
                pmax = p
            end
        end
    end
    pmax
end
println(search())

File: Scala/problem004.scala
// Problem 4 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=4

object Problem003 {
  def isPalindromic(x: Int): Boolean = {
    x.toString.reverse == x.toString
  }

  def main(args: Array[String]) :Unit = {
    val pmax = (100 until 1000)
      .flatMap(i => (i until 1000).map(j => i * j))
      .toSet
      .filter(isPalindromic)
      .max

    println(pmax)
  }
}
